### ISSUE TITLE
`:discouraged-namespace` does not warn on namespaced keywords usage without require

### ISSUE THREAD

**Issue by rap1ds** [clj-kondo/clj-kondo#2656]
- [X] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

```
➜  clj-kondo --version
clj-kondo v2025.10.23
```

**platform**

macOS, installed using the [installation script](https://github.com/clj-kondo/clj-kondo/blob/master/doc/install.md#installation-script-macos-and-linux)

**problem**

The documentation for discouraged namespace says: "warn on the require or usage of a namespace that is discouraged to be used." However, if I use a namespaced keyword without requiring it, I don't get a warning.

**repro**

✅ ns require works as expected:

```
➜  echo '(ns com.my.core (:require [com.my.discouraged]))' | clj-kondo --repro --config '{:linters {:discouraged-namespace {com.my.discouraged {:message "com.my.discouraged is discouraged"}}}}' --lint -

<stdin>:1:28: warning: com.my.discouraged is discouraged
linting took 52ms, errors: 0, warnings: 1
```

Expected: warns about discouraged namespace

Actual: warns about discouraged namespace

❌ Namespaced keyword usage without require does not work as expected:

```
➜  echo '(ns com.my.core) {:com.my.discouraged/foo "foo"}' | clj-kondo --repro --config '{:linters {:discouraged-namespace {com.my.discouraged {:message "com.my.discouraged is discouraged"}}}}' --lint -

linting took 50ms, errors: 0, warnings: 0
```

Expected: warns about discouraged namespace

Actual: No warning

**Comment by borkdude** [comment]
The reason for this is. that the keyword doesn't depend on the namespace to exist or loaded in the consuming namespace.
